src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CDAConfiguration.java

```diff
@@ -98,4 +98,20 @@ public Optional<URI> getPrivateKeyUri() {
     public Optional<URI> getCertificateUri() {
         return ca.getCertificateUri();
     }
+
+    /**
+     * Verifies if the configuration for the certificateAuthority has changed, given a previous
+     * configuration.
+     *
+     * @param config  CDAConfiguration
+     */
+    public boolean certificateAuthorityChanged(CDAConfiguration config) {
+        if (config == null) {
+            return true;
+        }
+
+        return !config.getCertificateUri().equals(getCertificateUri())
```

Comment:
You can use `Objects.equals` to handle nulls better

Reply:
Will do